### DISCUSSION TITLE
Incomplete CSS breadcrumbs

### DISCUSSION THREAD

**Discussion by erosman** [mdn/mdn-community#223] [Platform] [answered]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/background-size

### What specific section or headline is this issue about?

breadcrumbs

### What information was incorrect, unhelpful, or incomplete?

Most (if not all) of the CSS section (unlike JavaScript section) lacks proper breadcrumbs and the higher section goes to the [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS) which doesn't have links to the indexed article.

[References](https://developer.mozilla.org/en-US/docs/Web) > [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS) > [background-size](https://developer.mozilla.org/en-US/docs/Web/CSS/background-size)

The consequence is that in order to go from `background-size` to `background`, one has to perform a search.

### What did you expect to see?

For example:

[References](https://developer.mozilla.org/en-US/docs/Web) > [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS) > [CSS reference](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference) > [background](https://developer.mozilla.org/en-US/docs/Web/CSS/background) > [background-size](https://developer.mozilla.org/en-US/docs/Web/CSS/background-size)

**Comment by wbamberg** [comment]
This is really tricky.

The proximate problem here is that breadcrumbs are AFAIK derived from the URL structure, and in the CSS docs (almost) all pages are directly under /Web/CSS. So to fix this we'd need to reorganize all the page under CSS.

But the bigger problem is: what kind of organization do we want? The report says:

> For example:
>
> [References](https://developer.mozilla.org/en-US/docs/Web) > [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS) > [CSS reference](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference) > [background](https://developer.mozilla.org/en-US/docs/Web/CSS/background) > [background-size](https://developer.mozilla.org/en-US/docs/Web/CSS/background-size)

...but as you say, that's just an example. What would be needed here would be a complete IA for the CSS docs, that assigns a place in the hierarchy to everything, and it's not that clear what this could be.

@Rumyra , is there any work planned to reorg the CSS docs?

**Comment by erosman** [comment]
For example W3SChools has a fairly reasonable set-up e.g. [CSS Backgrounds](https://www.w3schools.com/css/css_background.asp)

The layout is:

- CSS Backgrounds
  - Background Color
  - Background Image
  - Background Repeat
  - Background Attachment
  - Background Shorthand

To start with, I would definitely suggest putting the CSS under [CSS reference](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference). At least that would make it easier to find items.

Next, as many CSS have shorthand declaration, the shorthand would be the parent, as in above example.

Next, the more difficult one is the hierarchy of related items e.g. `top, bottom, left, right`, coming under `position`.

Lastly, other groups like [Pseudo-classes](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes) which are already organized.

**Comment by teoli2003** [comment]
Note that this also would have an impact on sidebars et on macros. This is a 6-8 month project with significant dev support.

The first step is likely to define the IA as @wbamberg said, there are about 800-900 pages here. How many levels, how does it relate to the overview pages, where to place guides about one module, about several modules?

Once this is defined, there one need to study what has to be done in Yari, in the sidebars, in the breadcrumb, in the macro, and the content (Each page has a _See also_ section that will need to be updated for the new structure.

Finally, there will be a transition plan, as this will not likely happen overnight.

Yes, definitively at least half a year of work.

**Comment by Rumyra** [comment]
> @Rumyra , is there any work planned to reorg the CSS docs?

@wbamberg not at this time

I agree with a number of comments above: This is a big project and needs careful consideration. I'm going to transfer this to a discussion and bring it up in the editorial 👍

**Comment by Rumyra** [comment, marked as answer]
We have reorganised the CSS directory and the breadcrumbs now reflect that
